tests/Unit/Speed/Knot.php

```diff
@@ -0,0 +1,53 @@
+<?php
```

Comment:
The reason the pipeline is failing on a PSR4 error: The name of this file should be `KnotTest.php` instead of `Knot.php`.

Reply:
I think is also the reason for the above comment about lines not being covered: PHPUnit by defaults only executes test in classes ending with "Test.php", so if this is renamed the coverage for this file should also be fixed. ;)

Reply:
Oh dear. Very sorry about this. I've renamed the file on my local copy of my branch in my fork. Git is not letting me commit it saying I need to pull and when I try to pull, it says I can't do that. 🤷🏼

Reply:
If you undo all the local changes, will it let you pull? Otherwise deleting the local version of this branch and checking out the remote version might fix things.  If that doesn't help I can also rename the file. ;)

Reply:
I'm really sorry. I am totally confused. If you can fix this on your end please please do so. I'll then delete my fork entirely. (I knew this seemed too easy! 😅)

Reply:
Hm, I may have just fixed it? My fork and branch are not up to date…

Reply:
Yeah, you did it! I updated your branch for you when updating from main. When you open a PR from a fork by default the "allow edits by maintainers" checkbox is enabled, which allows me to update the branch in your repository, which is what I did before. ;)

Reply:
Great. Hah. 
Looks like al the checks pass now 👍🏼